hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/AbstractHoodieWriteClient.java

```diff
@@ -203,6 +208,45 @@ public boolean commitStats(String instantTime, List<HoodieWriteStat> stats, Opti
     return true;
   }
 
+  public boolean commitStats(String instantTime, List<HoodieWriteStat> stats, Option<Map<String, String>> extraMetadata,
+                             String commitActionType, Map<String, List<String>> partitionToReplaceFileIds) {
+
+    HoodieCommitMetadata metadata = CommitUtils.buildMetadata(stats, partitionToReplaceFileIds, extraMetadata,
+        operationType, config.getSchema(), commitActionType);
+    if (config.isMultiWriterEnabled()) {
```

Comment:
use the concurrency mode config introduced by the other PR and use that instead to fence this block?

Reply:
That's the plan, once it's landed will change this, for now, this config is just a place holder.